### DISCUSSION TITLE
features issue

### DISCUSSION THREAD

**Discussion by BuckeyePay** [Chimildic/goofy#263] [Помощь с ошибкой] [answered]
I have a script that selects a range of tracks that are three minutes in length. The script has stopped working and returns a bunch of 403 errors. If I remove the features component, it runs fine.

This is the offending command.

  Filter.rangeTracks(tracks, {
    features: {
      duration_ms: { min: 180000, max: 240000 }
    }
  });

**Comment by Chimildic** [comment, marked as answer]
Spotify has updated [API policies](https://developer.spotify.com/blog/2024-11-27-changes-to-the-web-api). Features like `danceability`,`valence` and so on have been deprecated. The `duration_ms` is part of features request. 

Change `features` to `meta`. The track also has duration property.
```js
Filter.rangeTracks(tracks, {
    meta: {
        duration_ms: { min: 180000, max: 240000 }
    }
})
```

**Reply by hahahahywel** [reply]
Is there any workaround for tempo?

**Reply by Chimildic** [reply]
@hahahahywel no, maybe later. I am trying to find solution.

**Reply by Chimildic** [reply]
@hahahahywel check [migration guide](https://chimildic.github.io/goofy/#/migrate2)

**Reply by hahahahywel** [reply]
Thanks very much I'll give it a go later :D

**Reply by hahahahywel** [reply]
> @hahahahywel check [migration guide](https://chimildic.github.io/goofy/#/migrate2)

Got it working thanks for your help!

**Comment by BuckeyePay** [comment]
Thank you.
